### PR TITLE
fixed flow update bug

### PR DESCRIPTION
**Pull request by irritanterik**
fixed issue #4

### CODE REVIEW

File: lib/folderFlows.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ function updateFolderFlows (folderId, which, enabled, callback) {
     }
     flows.forEach(flow => {
       count++
-      updateFlowEnabled(flow.id, enabled)
+      updateFlowEnabled(flow, enabled)
```
Comment (matjaz) [inline lib/folderFlows.js]: thanks for debugging this.
I would change `getFolderFlows` so it returns flow not ids. As it was and name implies. Do you mind making that change. I am not able to do that atm.
Also please bump app version.

Thank you!

**Review by matjaz — CHANGES_REQUESTED** [review]

File: lib/folderFlows.js (hunk 1 above)
Comment (irritanterik) [reply lib/folderFlows.js]: will look into it.

File: lib/folderFlows.js [hunk 2]
```diff
@@ -100,8 +100,12 @@ function getFolderFlows (folderId, which) {
   ]).then(result => {
     const flows = result[0]
     const folderIds = result[1]
+    var flowsFiltered = []
     folderIds.push(folderId)
-    return Object.keys(flows).filter(flowId => folderIds.indexOf(flows[flowId].folder) !== -1)
+    Object.keys(flows)
+    .filter(flowId => folderIds.indexOf(flows[flowId].folder) !== -1)
+    .forEach(flowId => { flowsFiltered.push(flows[flowId]) })
```
Comment (matjaz) [inline lib/folderFlows.js]: please use `.map`

**Comment by irritanterik** [comment]
Done.

**Review by matjaz — CHANGES_REQUESTED** [review]

**Comment by irritanterik** [comment]
Done. And finally understand .map 👍

File: lib/folderFlows.js [hunk 3]
```diff
@@ -100,12 +100,9 @@ function getFolderFlows (folderId, which) {
   ]).then(result => {
     const flows = result[0]
     const folderIds = result[1]
-    var flowsFiltered = []
-    folderIds.push(folderId)
```
Comment (matjaz) [inline lib/folderFlows.js:103]: this should not be removed.

**Review by matjaz — CHANGES_REQUESTED** [review]

File: lib/folderFlows.js [hunk 4]
```diff
@@ -100,8 +100,9 @@ function getFolderFlows (folderId, which) {
   ]).then(result => {
     const flows = result[0]
     const folderIds = result[1]
-    folderIds.push(folderId)
```
Comment (matjaz) [inline lib/folderFlows.js:103]: this should not be removed

**Review by matjaz — CHANGES_REQUESTED** [review]

**Comment by matjaz** [comment]
filter/map/reduce is pretty powerfull

**Comment by matjaz** [comment]
Thank you and sorry for nagging :}

**Comment by irritanterik** [comment]
No problem. Best way to learn it the good way (and prevent bugs).

**Pull request merged**